src/components/application_manager/src/help_promt_manager.cc

```diff
@@ -0,0 +1,418 @@
+  }
+}
+
+void HelpPromtManager::onTimerExpired() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  for (auto& aCommand : vr_commands_) {
+    count_timer_commands += aCommand.second->length();
+  }
+  OnTimeoutExpired();
+}
+
+void HelpPromtManager::SendTTSRequest() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (!vr_commands_.size()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+  if (!app_manager_) {
+    LOG4CXX_DEBUG(logger_, "Application Manager isn't initialized");
+    return;
+  }
+  LOG4CXX_DEBUG(logger_, "for appID:" << app_.app_id());
+  smart_objects::SmartObjectSPtr tts_global_properties =
+      utils::MakeShared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map);
+  if (tts_global_properties) {
+    smart_objects::SmartObject& ref = *tts_global_properties;
+
+    ref[strings::params][strings::message_type] =
+        static_cast<int>(hmi_apis::messageType::request);
+    ref[strings::params][strings::protocol_version] =
+        commands::CommandImpl::protocol_version_;
+    ref[strings::params][strings::protocol_type] =
+        commands::CommandImpl::hmi_protocol_type_;
+    ref[strings::params][strings::correlation_id] =
+        app_manager_->GetNextHMICorrelationID();
+
+    smart_objects::SmartObject& so_to_send = *tts_global_properties;
+    so_to_send[strings::params][strings::function_id] =
+        static_cast<int>(hmi_apis::FunctionID::TTS_SetGlobalProperties);
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    CreatePromtMsg(msg_params);
+
+    msg_params[strings::app_id] = app_.app_id();
+    so_to_send[strings::msg_params] = msg_params;
+    app_manager_->ManageHMICommand(tts_global_properties);
+  }
+}
+
+void HelpPromtManager::SendUIRequest() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (!vr_commands_.size()) {
+    LOG4CXX_DEBUG(logger_, "vr_commands_ is empty");
+    return;
+  }
+  if (!app_manager_) {
+    LOG4CXX_DEBUG(logger_, "Application Manager isn't initialized");
+    return;
+  }
+  LOG4CXX_DEBUG(logger_, "for appID:" << app_.app_id());
+  smart_objects::SmartObjectSPtr ui_global_properties =
+      utils::MakeShared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map);
+  if (ui_global_properties) {
+    smart_objects::SmartObject& ref = *ui_global_properties;
+
+    ref[strings::params][strings::message_type] =
+        static_cast<int>(hmi_apis::messageType::request);
+    ref[strings::params][strings::protocol_version] =
+        commands::CommandImpl::protocol_version_;
+    ref[strings::params][strings::protocol_type] =
+        commands::CommandImpl::hmi_protocol_type_;
+    ref[strings::params][strings::correlation_id] =
+        app_manager_->GetNextHMICorrelationID();
+
+    smart_objects::SmartObject& so_to_send = *ui_global_properties;
+    so_to_send[strings::params][strings::function_id] =
+        static_cast<int>(hmi_apis::FunctionID::UI_SetGlobalProperties);
```

Comment:
@SNiukalov as I remember you don't need `static_cast` for `SmartObject` inisitalization